app/javascript/components/GuitarGallery.test.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { render, screen /*, waitForElement*/ } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import axios from 'axios';
+import GuitarGallery from './GuitarGallery';
+import { guitarsUrl } from './constants';
+import { guitarData } from '../../__fixtures__/GuitarsData';
+
+test("show loader when it's fetching data, then render Guitars", async () => {
+  axios.get.mockResolvedValueOnce(guitarData);
+  //show loader
+  const { unmount /*, getAllByTestId, getByText*/ } = render(<GuitarGallery />);
+  // expect(getByText(/loading.../i)).toBeInTheDocument();
+
+  // check the correct url is called:
+  expect(axios.get).toHaveBeenCalledWith(guitarsUrl);
+  expect(axios.get).toHaveBeenCalledTimes(1);
+
+  //   //check what's rendered in the row
+  //  const rowValues = await waitForElement(() => getAllByTestId('row').map((row) => row.textContent));
+  //  expect(rowValues).toEqual(['ali', 'abu']);
+
+  const guitarDescriptions = await screen.findAllByText(/Description/i);
+  // const guitarDescriptions = screen.findAllBy('Description');
+  const expectedDescription = [
+    'some description in guitar1_test',
+    'some description in guitar2_test'
+  ];
+  // option 1:
+  expect(guitarDescriptions).toEqual(expectedDescription);
```

Comment:
I think you need to test if each string exists in the document instead of comparing an array of strings to the DOM objects which contain those strings. 

There must be a react-testing-library assertion that checks if a string exists in the DOM

Reply:
OK I'll have a look 
Thanks Ron 🙏🏼

Reply:
Fixed 👍🏼